lib/modules/dosomething/dosomething_user/dosomething_user_social_auth.inc

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * Constructs a new Drupal user for the given account details to represent a Facebook user.
+ * @param  array $account_details  All of the properties to instantiate
+ * @return $user
+ */
+function dosomething_user_make_facebook_user($account_details) {
+  $edit = [
+    'mail' => $account_details['email'],
+    'name' => $account_details['email'],
+    'pass' => user_password(),
+    'status' => 1,
+    'created' => REQUEST_TIME,
+  ];
+
+  dosomething_user_set_fields($edit, $account_details);
+
+  $user = user_save(null, $edit);
+  user_save($user, ['name' => $user->uid]);
+
+  // For simplicity we don't need to attach this before the save / save it to a table,
+  // it just needs to be here for object transformations
+  $user->facebook_id = $account_details['facebook_id'];
+  
+  return $user;
+}
+
+/**
+ * Registers a new user with Northstar (or upserts with existing) & returns a Drupal account
+ * @param  array $account_details Account details for the user to register
+ * @return $user Created (or retrieved) Drupal user
+ */
+function dosomething_user_register_facebook_user($account_details) {
+  // Otherwise check if we need to make a new account or upsert with an existing.
+  $create_new_account = TRUE;
+
+  // Checks if there is an existing account for the given email
+  if (isset($account_details['email'])) {
+    $email = $account_details['email'];
+    $create_new_account = !dosomething_northstar_user_exists_by_index('email', $email);
+  }
+
+  if ($create_new_account) {
+    // Create the Drupal user.
+    // Unfortunately there doesn't seem to be anyway around this because of how Drupal is architected.
+    $user = dosomething_user_make_facebook_user($account_details);
+    dosomething_northstar_create_user($user);
+  }
+  else {
+    // Update existing Northstar user to link Facebook ID
+    $user->facebook_id = $account_details['facebook_id'];
```

Comment:
@DFurnes I was toying with adding this here, but then i realized this user is anonymous, so NS wont know what to do with it?